test/puppetlabs/rbac_client/services/test_activity.clj

```diff
@@ -1,6 +1,115 @@
+    :id "42bf351c-f9ec-40af-84ad-e976fec7f4bd"}
+   :object {:name "Herman Aldrich"
+            :id "2d5d4e0b-5353-4bbd-9c4c-084177caac32"
+            :type "a Herman"}})
+
+(def expected-v1-bundle
+  {:commit
+   {:service
+    {:id "namer"}}
+   :subject
+   {:name "Herman Melville"
+    :id "42bf351c-f9ec-40af-84ad-e976fec7f4bd"}
+   :object{:name "Herman Aldrich"
+           :id "2d5d4e0b-5353-4bbd-9c4c-084177caac32"
+           :type "a Herman"}})
+
+(def expected-v1-upgraded-bundle
+  {:commit
+   {:service
+    {:id "namer"}}
+   :subject
+   {:name "Herman Melville"
+    :id "42bf351c-f9ec-40af-84ad-e976fec7f4bd"}
+   :objects [{:name "Herman Aldrich"
+              :id "2d5d4e0b-5353-4bbd-9c4c-084177caac32"
+              :type "a Herman"}]})
+
+
+(deftest test-activity
+  (with-app-with-config tk-app [remote-activity-reporter] (:client configs)
+      (let [consumer-svc (tk-app/get-service tk-app :ActivityReportingService)
+            handler (wrap-test-handler-middleware
+                             (fn [req]
+                               (if (= "/activity-api/v2/events" (:uri req))
+                                 (let [parsed-body (json/parse-string (:body req) true)]
+                                   (is (or (= v2-bundle parsed-body) (= expected-v1-upgraded-bundle parsed-body)))
+                                   ;; provide a payload for testability.  Actual service doesn't guarantee providing return results
+                                   (http/json-200-resp {:success true :actual (:body req)}))
+                                 (is false))))
+            v1-handler (wrap-test-handler-middleware
+                        (fn [req]
+                          (case (:uri req)
+                            "/activity-api/v2/events" (http/json-resp 404 {:success false})
+                            "/activity-api/v1/events" (http/json-200-resp {:success true :actual (:body req)}))))]
+
+        (testing "v2 endpoint supported with passthrough"
+          (with-test-webserver-and-config handler _ (:server configs)
+            (let [result (act/report-activity! consumer-svc v2-bundle)]
+              (is (= 200 (:status result)))
+              (is (= true (get-in result [:body :success])))
+              (is (= v2-bundle (json/parse-string (get-in result [:body :actual]) true))))))
+
+        (testing "upgrades to v2 payload when v1 payload is submitted"
+          (with-test-webserver-and-config handler _ (:server configs)
+                                          (let [result (act/report-activity! consumer-svc v1-bundle)]
+                                            (is (= 200 (:status result)))
+                                            (is (= true (get-in result [:body :success])))
+                                            (is (= expected-v1-upgraded-bundle (json/parse-string (get-in result [:body :actual]) true))))))
+
+        ;; note that after this test, the caching for the v1 endpoint is in place
```

Comment:
hey Jon, thanks for the helpful comments throughout here. A nit but it took me a little while to figure out what you meant by "the caching for the v1 endpoint" in this comment- I think you're referencing that the value of :supports-v2-api will be set false after this test runs, is that right? If so, could be worth making that even more explicit in case anyone else reading this is similarly confused.

Reply:
Yep.  Will do.

Reply:
Great, that's much clearer, thanks!